gcovr/tests/filter-relative-lib-from-unfiltered-tracefile/Makefile

```diff
@@ -9,7 +11,11 @@ all: links
 run: txt xml html sonarqube
 
 # the src/ filter is provided by the project/gcovr.cfg configuration file
+ifeq ($(filter $(BASE_OS),MSYS_NT MINGW64_NT),)
 GCOVR_TEST_OPTIONS = -f '\.\./external-library/src'
+else
+GCOVR_TEST_OPTIONS = -f '.*/external-library/src'
+endif
```

Comment:
The junction internally resolves the relative path. If you list the directory contents the path is shown absolute.

Reply:
Yes, but gcovr's filter turns it into a relative path for regex matching, right? So if the current working directory is `C:/gcovr/tests/foo/project` and the realpath of the `./relevant-library` is `C:/gcovr/tests/foo/external-library`, I'd expect the following call to succeed:

```python
filter = RelativeFilter(root="C:/gcovr/tests/foo/project",
                        pattern=r"\.\./external-library/")
assert filter.match("./relevant-library/src/foo.c")
```

If not, that would be a bug with how the filters are implemented and the test would still be broken.

Reply:
Tnan we have a bug here. I'll investigate into this the next days.

Reply:
To be clear, I suspect that your new code is fine. But if the tests didn't work with the same filter pattern for all OSes, then the filter implementation is likely broken for Windows symlinks. I guess replacing the `os.path.realpath()` calls with your `gcovr_realpath()` might do the trick, but ultimately it could be that these things will only start working once this PR + #535 are combined.

Potential path forward: use the same GCOVR_TEST_OPTIONS for all OSes in this and the other test, but mark these tests as xfail on Windows. Merge, then update #535 and see if things start working. Might need another iteration for everything to fall into place. Caveat: I'd be a bit unhappy about releasing a gcovr-5.1 with half-baked Windows symlink support, on the other hand we'd have to wait forever until the path handling logic in `gcov.py` gets truly untangled.

Reply:
I'll use the same value for all OSes and try to get the tests working.

The date for the 5.1 was only a rough estimation, of cours I'll wait for all known bugs to be fixed.

Reply:
I've found the problem. We detect absolute filters by the `os.path.isabs`:
https://github.com/gcovr/gcovr/blob/6a79a04d15959adfa87af27bd384dccdcc7dac18/gcovr/utils.py#L181-L184

The problem is that the function uses the path seperator of the OS. Verbose output on Ubuntu:
```
- RelativeFilter(src/ root=)
- RelativeFilter(\.\./external-library/src root=/gcovr/gcovr/tests/filter-relative-lib-from-unfiltered-tracefile/project)
```
and on Windows:
```
- RelativeFilter(src/ root=)
- AbsoluteFilter(\.\./external-library/src)
```

Reply:
This is the kind of bug that is absolutely obvious *in hindsight*. Thank you for finding and fixing this!!!